src-docs/src/views/form_controls/guidelines.js

```diff
@@ -0,0 +1,272 @@
+        </GuideRuleExample>
+      </GuideRule>
+
+      <EuiSpacer />
+
+      <EuiTitle>
+        <h2>Hint text</h2>
+      </EuiTitle>
+
+      <GuideRule
+        description={
+          <p>
+            Place hints and instructions outside a text field so it is always
+            visible to the user. Use hint text to:
+            <EuiSpacer />
+            <ul>
+              <li>
+                Explain <strong>why you are asking a certain question</strong>:
+                &quot;We will only email you if there is a problem with your
+                order.&quot;
+              </li>
+              <li>
+                Provide <strong>clarifying details</strong> on what to type:
+                &quot;Enter the full 32-characters ID.&quot;
+              </li>
+              <li>
+                Tell the user <strong>where to find the information</strong>{' '}
+                you&apos;re asking for: &quot;Find the Elasticsearch cluster ID
+                on the main administration page of your deployment.&quot;
+              </li>
+            </ul>
+          </p>
+        }
+      >
+        <GuideRuleExample
+          panelDisplay="block"
+          type="do"
+          text="Help users make the right decision by clarifying what goes inside a field."
+        >
+          <EuiFormRow
+            label="Timeout"
+            helpText="Use shorthand notation, such as 30s, 10m, or 1h."
+          >
+            <EuiFieldText />
+          </EuiFormRow>
+        </GuideRuleExample>
+
+        <GuideRuleExample
+          panelDisplay="block"
+          type="do"
+          text="Use complete sentences and ending punctuation."
+        >
+          <EuiFormRow
+            label="Name"
+            helpText="This is the name that will be visible to other users."
+          >
+            <EuiFieldText />
+          </EuiFormRow>
+        </GuideRuleExample>
+      </GuideRule>
+
+      <EuiSpacer />
+
+      <EuiTitle>
+        <h2>Placeholder text</h2>
+      </EuiTitle>
+
+      <GuideRule
+        heading="In fields"
+        description={
+          <p>
+            Use the placeholder property to describe the expected value of the
+            input or to provide a useful example.
+          </p>
+        }
+      >
+        <GuideRuleExample
+          panelDisplay="block"
+          type="do"
+          text="Provide a simple example that highlights the expected syntax or value."
+        >
+          <EuiFormRow
+            label="Phone number"
+            helpText="Include the international call prefix."
+          >
+            <EuiFieldText placeholder="+33600000000" />
+          </EuiFormRow>
+        </GuideRuleExample>
+        <GuideRuleExample
+          panelDisplay="block"
+          type="dont"
+          text="Start with `for example`, or `e.g.`."
+        >
+          <EuiFormRow label="Endpoint" helpText="Include the port number.">
+            <EuiFieldText placeholder="e.g. https://example.com:9500" />
```

Comment:
Drive by comment: In the Writing Style Guide, we're advised not to use "e.g.": https://brand.elastic.co/302f66895/p/194a3b-writing-style-guide. Should we use this instead?
```suggestion
            <EuiFieldText placeholder="For example, https://example.com:9500" />
```